crates/bestool/src/actions/crypto/encrypt.rs

```diff
@@ -0,0 +1,39 @@
+use std::path::PathBuf;
+
+use age::x25519;
+use clap::Parser;
+use miette::{miette, Context as _, IntoDiagnostic as _, Result};
+
+use crate::actions::{crypto::CryptoArgs, Context};
+
+#[derive(Debug, Clone, Parser)]
+pub struct EncryptArgs {
+	plaintext: PathBuf,
+
+	#[arg(long)]
+	public_key: PathBuf,
+}
+
+pub async fn run(ctx: Context<CryptoArgs, EncryptArgs>) -> Result<()> {
+	let public_key: x25519::Recipient = tokio::fs::read_to_string(&ctx.args_sub.public_key)
+		.await
+		.into_diagnostic()
+		.wrap_err("reading the public key")?
+		.parse()
+		.map_err(|err: &str| miette!("failed to parse: {err}"))?;
+
+	let plaintext = tokio::fs::read(&ctx.args_sub.plaintext)
```

Comment:
we'll need to handle files larger than memory, so some kind of streaming or iterative encryption is needed here (and same with decrypting)